xds/internal/balancer/clusterresolver/resource_resolver_dns.go

```diff
@@ -116,6 +120,10 @@ func (dr *dnsDiscoveryMechanism) stop() {
 // updates from the real DNS resolver.
 
 func (dr *dnsDiscoveryMechanism) UpdateState(state resolver.State) error {
+	if dr.logger.V(2) {
+		dr.logger.Infof("DNS discovery mechanism for resource %q reported an update: %s", dr.target, pretty.ToJSON(state))
```

Comment:
Optional Nit: Does pretty.json work well with the whole resolver state? Should this be pretty.JSON(state.ServiceConfig) and another logging way for Addresses and Attributes?

Reply:
This would end up as something like:

```
        {
          "Addresses": [
            {
              "Addr": "127.0.0.1:62414",
              "ServerName": "",
              "Attributes": {},
              "BalancerAttributes": null,
              "Type": 0,
              "Metadata": null
            },
            {
              "Addr": "127.0.0.1:62415",
              "ServerName": "",
              "Attributes": {},
              "BalancerAttributes": null,
              "Type": 0,
              "Metadata": null
            } 
          ],
          "ServiceConfig": null,
          "Attributes": null
        }
```

So, it does print all of it. But I don't know if `ServiceConfig` would render properly under this scheme. I'm pretty sure the addresses and attributes do, which is what we care about most of the time.

Reply:
Ah ok sounds good :).

Reply:
I approved this btw, feel free to merge.